packages/sanity/src/core/templates/resolve.ts

```diff
@@ -189,7 +189,7 @@ export async function resolveInitialObjectValue<Params extends Record<string, un
 
   const merged = deepAssign(fieldValues, initialObject)
 
-  if (isEmpty(merged)) {
+  if (isDeepEmpty(merged)) {
```

Comment:
I'm not 100% sure we need to do a deep emptyness check here, but leaving it as-is just to be sure